src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -96,143 +103,211 @@ public class FromBase64Transform : ICryptoTransform
     {
         private byte[] _inputBuffer = new byte[4];
         private int _inputIndex;
-        private FromBase64TransformMode _whitespaces;
+        private readonly FromBase64TransformMode _whitespaces;
 
         public FromBase64Transform() : this(FromBase64TransformMode.IgnoreWhiteSpaces) { }
         public FromBase64Transform(FromBase64TransformMode whitespaces)
         {
             _whitespaces = whitespaces;
-            _inputIndex = 0;
         }
 
         // Converting from Base64 generates 3 bytes output from each 4 bytes input block
-        public int InputBlockSize => 1;
+        public int InputBlockSize => 4;
         public int OutputBlockSize => 3;
         public bool CanTransformMultipleBlocks => false;
         public virtual bool CanReuseTransform => true;
 
         public int TransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, byte[] outputBuffer, int outputOffset)
         {
-            ValidateTransformBlock(inputBuffer, inputOffset, inputCount);
-            if (_inputBuffer == null) throw new ObjectDisposedException(null, SR.ObjectDisposed_Generic);
+            // inputCount != InputBlockSize is not allowed
```

Comment:
I don't see that requirement from .NET Framework, suggesting this is a breaking change.

Reply:
I have the assumption that `TransformBlock` transforms `InputBlockSize` to `OutputBlockSize`.
Based on that assumption the code is optimized. E.g. https://github.com/dotnet/corefx/blob/6087a2c79e0aceeae8dc96036c6809ece093c996/src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs#L131-L132

In the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.frombase64transform.transformblock) there's no clear statement how big `inputCount` can be. The doc state rather generic

> ArgumentException
> `inputCount` uses an invalid value. 

And

```c#
// Iterate through inputBytes transforming by blockSize.
int inputBlockSize = base64Transform.InputBlockSize;
```

So based on the above assumption, symmetry with `ToBase64Transform`, and on these info from the docs this change was made.

Should any (valid) `inputCount` be allowed? Or is this change -- while beeing a breaking change -- OK?

---

Maybe the docs should be precizied on this?

Reply:
It will depend on the behaviour with <4 input size. If it buffers the input until there is enough transform and then reports errors in TransformFinalBlock then any input size is valid here and output must be a multiple of 3.

Reply:
> If it buffers the input until there is enough transform 

That's the case. 
But in my point of view is this buffering only necessary due to the whitespace-handling. If there would be no whitespace, and no handling for it, there could be 4 input bytes decoded to 3 output bytes.

From a consumer side: why would one call `TransformBlock` with `inputCount != InputBlockSize`?
(I know this is a very weak argument for enforcing this.)

Reply:
>From a consumer side: why would one call TransformBlock with inputCount != InputBlockSize? 

Because weird things happen and sometimes you don't get aligned input that you're passing through to the decoder. imagine someone passing you a string which is base64 prefixed with some other data for example. Or uncompressing chars and passing them through to a decoder stream.

Reply:
Given how old this type is, it matters more what it has historically done than what it "should" do.  Aside from the perf numbers, this change should (ideally) be invisible from a black-box perspective.  Exceptions don't change, including that all previous inputs (no matter how little sense they made) produce the output they used to.